src/k8s/pkg/k8sd/api/certs_refresh.go

```diff
@@ -0,0 +1,254 @@
+			organization: []string{"system:nodes"},
+			usages:       []v1.KeyUsage{v1.UsageDigitalSignature, v1.UsageKeyEncipherment, v1.UsageClientAuth},
+			signerName:   "k8sd.io/kubelet-client",
+		},
+		{
+			name:       fmt.Sprintf("k8sd-%d-worker-kube-proxy-client", req.Seed),
+			commonName: "system:kube-proxy",
+			usages:     []v1.KeyUsage{v1.UsageDigitalSignature, v1.UsageKeyEncipherment, v1.UsageClientAuth},
+			signerName: "k8sd.io/kube-proxy-client",
+		},
+	} {
+		csr := csr
+		g.Go(func() error {
+			componentCSR, pKey, err := pkiutil.GenerateCSR(
+				pkix.Name{
+					CommonName:   csr.commonName,
+					Organization: csr.organization,
+				},
+				2048,
+				nil,
+				csr.hostnames,
+				csr.ips,
+			)
+			if err != nil {
+				return fmt.Errorf("failed to generate CSR for %s: %w", csr.name, err)
+			}
+
+			_, err = client.CertificatesV1().CertificateSigningRequests().Create(errGroupCTX, &certificatesv1.CertificateSigningRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: csr.name,
+				},
+				Spec: certificatesv1.CertificateSigningRequestSpec{
+					Request:    []byte(componentCSR),
+					Usages:     csr.usages,
+					SignerName: csr.signerName,
+				},
+			}, metav1.CreateOptions{})
+
+			if err != nil {
+				return fmt.Errorf("failed to create CSR for %s: %w", csr.name, err)
+			}
+
+			for {
+				select {
+				case <-errGroupCTX.Done():
+					return nil
+				default:
+					request, err := client.CertificatesV1().CertificateSigningRequests().Get(errGroupCTX, csr.name, metav1.GetOptions{})
+					if err != nil {
+						log.Error(err, fmt.Sprintf("failed to get CSR %s", csr.name))
+						continue
+					}
+
+					approved, err := isCertificateSigningRequestApproved(request)
+					if err != nil {
+						log.Error(err, fmt.Sprintf("failed to check approval status for CSR %s", csr.name))
+						return fmt.Errorf("failed to check approval status for CSR %s: %w", csr.name, err)
+					}
+
+					if approved && isCertificateSigningRequestIssued(request) {
+						if _, _, err = pkiutil.LoadCertificate(string(request.Status.Certificate), ""); err != nil {
+							log.Error(err, fmt.Sprintf("failed to load certificate for CSR %s", csr.name))
+						}
+						switch csr.name {
+						case fmt.Sprintf("k8sd-%d-worker-kubelet-serving", req.Seed):
+							certificates.KubeletCert = string(request.Status.Certificate)
+							certificates.KubeletKey = string(pKey)
+						case fmt.Sprintf("k8sd-%d-worker-kubelet-client", req.Seed):
+							certificates.KubeletClientCert = string(request.Status.Certificate)
+							certificates.KubeletClientKey = string(pKey)
+						case fmt.Sprintf("k8sd-%d-worker-kube-proxy-client", req.Seed):
+							certificates.KubeProxyClientCert = string(request.Status.Certificate)
+							certificates.KubeProxyClientKey = string(pKey)
+						}
+						return nil
+					}
+				}
+			}
+		})
+
+	}
+
+	if err := g.Wait(); err != nil {
+		return response.InternalError(err)
+	}
+
+	log.Info("Ensuring worker PKI")
```

Comment:
log messages are the user's interface to see what is happening on k8sd. this message does not help them understand what is happening. either remove it, or increase verbosity `log.V(1).Info`, or adjust it in case there is useful information to convey

Reply:
same for other log messages